src/pages/index.js

```diff
@@ -7,10 +11,32 @@ import { ModalWithForm } from "../components/ModalWithForm.js";
 import { UserInfo } from "../components/UserInfo.js";
 import { ModalWithPhoto } from "../components/ModalWithPhoto.js";
 import { fillProfileForm } from "../utils/utils.js";
-
+import { Api } from "../components/Api.js";
 //imports config
 import { config } from "../utils/constants.js";
-//These lines add the function of for validation.
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "8213539a-e47c-4d36-92f9-050521f3ca6f",
+  },
+});
+
+let cardList;
+Promise.all([api.getInitialCards(), api.fetchProfile()])
+  .then(([cardsData, userData]) => {
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setUserAvatar(userData.avatar);
+    cardList = new Section(
+      { items: cardsData, renderer: createCard },
+      ".gallery"
+    );
+    cardList.renderItems();
+  })
+  .catch((error) => {
+    console.error("Oops, something happened", error);
+  });
+
 const editProfileFormValidator = new FormValidation(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```